charms/kfp-metadata-writer/src/charm.py

```diff
@@ -46,8 +45,10 @@ def __init__(self, *args):
         )
 
         self.grpc_relation = self.charm_reconciler.add(
-            component=SdiRelationDataReceiverComponent(
-                charm=self, name="relation:grpc", relation_name="grpc"
+            component=K8sServiceInfoComponent(
+                charm=self,
+                relation_name=GRPC_RELATION_NAME,
+                refresh_event=self.on[GRPC_RELATION_NAME].relation_changed,
```

Comment:
Do we also need `relation_broken` here?  I see it in the [SDI component](https://github.com/canonical/charmed-kubeflow-chisme/blob/8a7454fede8dbcefe65cd45f2fc04a3926b7e421/src/charmed_kubeflow_chisme/components/serialised_data_interface_components.py#L86).  I think yes, because if we remove the relation here the user wont be notified there is a problem

Reply:
Yeah, we do.  Tested manually and if I break a relation, I only see these events in the logs:
```
unit-kfp-metadata-writer-0: 15:31:47 INFO juju.worker.uniter.operation ran "grpc-relation-departed" hook (via hook dispatching script: dispatch)
unit-kfp-metadata-writer-0: 15:31:48 INFO juju.worker.uniter.operation ran "grpc-relation-broken" hook (via hook dispatching script: dispatch)
```
Currently, the charm wont know that it has lost its grpc service

Reply:
The library already handles that, I think what we need to do is observe both relation changed and relation broken events as suggested in your [previous comment](https://github.com/canonical/kfp-operators/pull/436#discussion_r1569449222).

Reply:
yeah I think I wrote this comment before [the above comment](https://github.com/canonical/kfp-operators/pull/436#discussion_r1569224916), and misunderstood why the `relation-broken` wasn't working.  We can close this

Reply:
@ca-scribner I have tested this in a live model and unfortunately, `relation-broken` is not observed(?). I will need some guidance on this. While adding the `self._events_to_observe` to observe the `relation-update` event, solves the issue with seeing the unit status go from blocked to active, removing the relation does not seem to trigger anything to go back from active to blocked.